sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/delete_key_operation.hpp

```diff
@@ -75,6 +75,14 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
         std::shared_ptr<Azure::Security::KeyVault::_internal::KeyVaultPipeline> keyvaultPipeline,
         Azure::Response<Azure::Security::KeyVault::Keys::DeletedKey> response);
 
+    DeleteKeyOperation(
+        std::shared_ptr<Azure::Security::KeyVault::_internal::KeyVaultPipeline> keyvaultPipeline,
```

Comment:
Shouldn't this take a XxxClient instead of a pipeline? Then you should just be able to call methods on the client.
It's not ideal to expose _internal things in public ctors.

Reply:
This is not a public cttor.
This is s private cttor.  Only a KeyVaultKey client can create `Operation<T>` , only the client has access to this cttor. 
There is an API in the client call `ResumeDeleteKey....` which takes the resume token and produces an operation.

The `Operation<T>` needs only the credential, pipeline and version to do polling. It does not need the client.

Reply:
Each call to Poll makes a service operation call just like any other service operation call. Normally, these call a method on the XxxClient and this should too; they should be logged, get request IDs, get retry support, and leverage any customer-added policies (all of these based on customer-set options).